src/cdk/scrolling/fixed-size-virtual-scroll.ts

```diff
@@ -121,11 +121,18 @@ export class FixedSizeVirtualScrollStrategy implements VirtualScrollStrategy {
     }
 
     const scrollOffset = this._viewport.measureScrollOffset();
-    const firstVisibleIndex = scrollOffset / this._itemSize;
     const renderedRange = this._viewport.getRenderedRange();
     const newRange = {start: renderedRange.start, end: renderedRange.end};
     const viewportSize = this._viewport.getViewportSize();
     const dataLength = this._viewport.getDataLength();
+    let firstVisibleIndex = scrollOffset / this._itemSize;
+
+    if (newRange.end >= dataLength) {
```

Comment:
Can you add a comment explaining why this is necessary?

Reply:
Comments added, I also changed the code to avoid unnecessary calculation in some cases.
Also added a test case with dynamic buffer size.